packages/react/src/hooks/useAuth.ts

```diff
@@ -28,7 +28,7 @@ export const useAuth = (): UseAuthResult => {
    * This util will be used to get current user after those events.
    */
   const fetchCurrentUser = React.useCallback(async () => {
-    setResult({ user: undefined, isLoading: true, error: undefined });
+    setResult((prevResult) => ({ ...prevResult, isLoading: true }));
```

Comment:
Why this change?

Reply:
Oh yeah, that needed some comment!

So this prevents `useAuth` from losing `user` value between `fetchCurrentUser` calls, by always spreading the previous value onto the new result.